extensions/amp-analytics/0.1/cookie-writer.js

```diff
@@ -224,13 +235,36 @@ export class CookieWriter {
         // provide a way to overwrite or erase existing cookie
         if (value) {
           const expireDate = Date.now() + cookieExpireDateMs;
+          // SameSite=None must be secure as per
+          // https://web.dev/samesite-cookies-explained/#samesitenone-must-be-secure
+          const secure = sameSite === SameSite.NONE;
           setCookie(this.win_, cookieName, value, expireDate, {
             highestAvailableDomain: true,
+            sameSite,
+            secure,
           });
         }
       })
       .catch((e) => {
         user().error(TAG, 'Error expanding cookie string', e);
       });
   }
+
+  /**
+   * Converts SameSite string to SameSite type.
+   * @param {string=} sameSite
+   * @return {SameSite|undefined}
+   */
+  getSameSiteType_(sameSite) {
+    switch (sameSite) {
```

Comment:
I think it's cleaner to do
```
return isEnumValue(SameSite, sameSite) ? sameSite : undefined;
```

Reply:
I agree. However, this approach means we're passing around strings as opposed to the SameSite enum values (defined in cookies.js) and I'd prefer not having strings floating around.